packages/tpetra/core/src/Tpetra_Details_crsUtils.hpp

```diff
@@ -112,37 +112,39 @@ pad_crs_arrays(
   if (additional_size_needed == 0)
     return;
 
+  using ptrs_value_type = typename RowPtr::non_const_value_type;
+  using inds_value_type = typename Indices::non_const_value_type;
+  using vals_value_type = typename Values::non_const_value_type;
+
   // The indices array must be resized and the row_ptr arrays shuffled
   auto indices_new = uninitialized_view<Indices>("ind new", indices.size()+additional_size_needed);
   auto values_new = uninitialized_view<Values>("val new", pad_values ? values.size()+additional_size_needed : 0);
-  Kokkos::deep_copy(values_new, Teuchos::ScalarTraits<typename Values::value_type>::zero());
+  Kokkos::deep_copy(values_new, vals_value_type(0.0));
 
   // mfh: Not so fussy about this not being a kernel initially,
   // since we're adding a new feature upon which existing code does not rely,
   // namely Export/Import to a StaticProfile CrsGraph.  However, watch out
   // for fence()ing relating to UVM.
   auto this_row_beg = row_ptr_beg(0);
   auto this_row_end = row_ptr_end(0);
```

Comment:
For the future: `Kokkos::deep_copy` now supports copying from a 0-D device View to a value in host memory.  Thus, you could do this:
```c++
ptrs_value_type this_row_beg;
Kokkos::deep_copy (this_row_beg, Kokkos::subview (row_ptr_beg, 0));
```

Reply:
I still have some work to do to build/pass tests on Cuda.  Is this preferred?  If so, I'll make the change now.

Reply:
@tjfulle I wouldn't worry about this for now.  You're accessing every row, so it will likely be faster just to let UVM do its thing.  I would prefer that we spent future effort on making this into a parallel kernel on device.  Thanks though!